### PR TITLE
fix: update exercise tracker test endpoints and test text

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Partially addresses https://github.com/freeCodeCamp/freeCodeCamp/issues/41529

This PR changes the test strings and test text for the URL Shortener project.

This needs to be tested and merged with #XXXXX in the [Demo Projects](https://github.com/freeCodeCamp/demo-projects) repo.

Here is a link to a live version of the project with the proposed changes in #XXXXX: https://exercise-tracker-update.scissorsneedfoo.repl.co

### CODE REVIEW

File: curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md [hunk 1]
```diff
@@ -66,12 +66,12 @@ async (getUserInput) => {
 };
 ```
 
-You can `POST` to `/api/exercise/add` with form data `userId=_id`, `description`, `duration`, and optionally `date`. If no date is supplied, the current date will be used. The response returned will be the user object with the exercise fields added.
+You can `POST` to `/api/users/{_id}/exercises` with form data `description`, `duration`, and optionally `date`. If no date is supplied, the current date will be used. The response returned will be the user object with the exercise fields added.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md:69]:
```suggestion
You can `POST` to `/api/users/:_id/exercises` with form data `description`, `duration`, and optionally `date`. If no date is supplied, the current date will be used. The response returned will be the user object with the exercise fields added.
```

Should we use the format we used for the params in the URL shortener PR?

File: curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md [hunk 2]
```diff
@@ -102,12 +102,12 @@ async (getUserInput) => {
 };
 ```
 
-You can make a `GET` request to `/api/exercise/log` with a parameter of `userId=_id` to retrieve a full exercise log of any user. The returned response will be the user object with a `log` array of all the exercises added. Each log item has the `description`, `duration`, and `date` properties.
+You can make a `GET` request to `/api/users/{_id}/logs` to retrieve a full exercise log of any user. The returned response will be the user object with a `log` array of all the exercises added. Each log item has the `description`, `duration`, and `date` properties.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md:105]:
```suggestion
You can make a `GET` request to `/api/users/:_id/logs` to retrieve a full exercise log of any user. The returned response will be the user object with a `log` array of all the exercises added. Each log item has the `description`, `duration`, and `date` properties.
```

File: curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md [hunk 3]
```diff
@@ -144,12 +144,12 @@ async (getUserInput) => {
 };
 ```
 
-A request to a user's log (`/api/exercise/log`) returns an object with a `count` property representing the number of exercises returned.
+A request to a user's log (`/api/users/{_id}/logs`) returns an object with a `count` property representing the number of exercises returned.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md:147]:
```suggestion
A request to a user's log (`/api/users/:_id/logs`) returns an object with a `count` property representing the number of exercises returned.
```

File: curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md [hunk 4]
```diff
@@ -185,12 +185,12 @@ async (getUserInput) => {
 };
 ```
 
-You can add `from`, `to` and `limit` parameters to a `/api/exercise/log` request to retrieve part of the log of any user. `from` and `to` are dates in `yyyy-mm-dd` format. `limit` is an integer of how many logs to send back.
+You can add `from`, `to` and `limit` parameters to a `/api/users/${_id}/logs` request to retrieve part of the log of any user. `from` and `to` are dates in `yyyy-mm-dd` format. `limit` is an integer of how many logs to send back.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md:188]:
```suggestion
You can add `from`, `to` and `limit` parameters to a `/api/users/:_id/logs` request to retrieve part of the log of any user. `from` and `to` are dates in `yyyy-mm-dd` format. `limit` is an integer of how many logs to send back.
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Small nitpick.

File: curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md (hunk 1 above)
Comment (scissorsneedfoodtoo) [reply curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/exercise-tracker.md:69]: Not a small nitpick at all! Thanks for catching that @nhcarrigan. I agree that we should be consistent with this.

I'll commit your suggestions now.

**Comment by ShaunSHamilton** [comment]
Just to mention: Whilst testing, I noticed the example Exercise Tracker app does not pass the 3rd test (current tests, with current app):
image

**Comment by nhcarrigan** [comment]
A quick note that the third test times out on the new project, though a manual check of the functionality suggests it is working.

I think we just have too much data in the database and it takes too long to query it all 😅 

image

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me.

**Comment by scissorsneedfoodtoo** [comment]
Thanks for your thorough review of this @ShaunSHamilton and @nhcarrigan. I noticed that during testing too. It probably doesn't help that the db is filled with test users from all the tests I ran :slightly_smiling_face:

**Comment by nhcarrigan** [comment]
It probably wouldn't hurt to dump that collection, honestly. I'll take a peek at the data in there tomorrow and see what everyone thinks about cleaning it.

**Comment by raisedadead** [comment]
> I'll take a peek at the data in there tomorrow and see what everyone thinks about cleaning it.

We should purge and re-seed all the data on the collections used for demo-projects. This should ideally be done on a daily basis with a CRON.

**Review by raisedadead — APPROVED** [review]

**Comment by nhcarrigan** [comment]
> We should purge and re-seed all the data on the collections used for demo-projects. This should ideally be done on a daily basis with a CRON.

With the exception of the URL shortener - IIRC Quincy wanted to preserve that data in case folks are actually using it.

**Pull request merged**